projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewService.java

```diff
@@ -189,7 +189,11 @@ protected List<ViewDTO> fetchViews() {
                             columns.add(new ColumnDTO(joinView.name + "_" + c.name, c.title, c.type, j.displayIndex));
                         }
                     }
-                    return new ViewDTO(v.name, v.title, columns);
+                    Long maxDisplayCount = viewsConfig
+                            .getViewConfig(v.name)
+                            .map(t -> t.maxDisplayCount)
+                            .orElse(null);
+                    return new ViewDTO(v.name, v.title, columns, maxDisplayCount);
```

Comment:
why retrieving maxDisplayCount again, if you have it available as `v.maxDisplayCount`?

Reply:
we extract the value if none null ViewConfig found by name

Reply:
but you are already iterating through views and you can have that value here from `v.maxDisplayCount`, can't you?

Reply:
```suggestion
                    return new ViewDTO(v.name, v.title, columns, v.maxDisplayCount);
```

Reply:
haha, indeed, facepalm

Reply:
applied